libr/core/panels.c

```diff
@@ -808,16 +808,47 @@ bool __check_panel_type(RPanel *panel, const char *type) {
 	if (!panel->model->cmd || !type) {
 		return false;
 	}
+	char *tmp = r_str_new (panel->model->cmd);
+	int n = r_str_split (tmp, ' ');
+	if (!n) {
+		return false;
+	}
+	const char *base = r_str_word_get0 (tmp, 0);
+	if (R_STR_ISEMPTY (base)) {
+		return false;
+	}
 	int len = strlen (type);
 	if (!strcmp (type, PANEL_CMD_DISASSEMBLY)) {
-		if (!strncmp (panel->model->cmd, type, len) &&
+		if (!strncmp (tmp, type, len) &&
 				strcmp (panel->model->cmd, PANEL_CMD_DECOMPILER) &&
 				strcmp (panel->model->cmd, PANEL_CMD_DECOMPILER_O) &&
 				strcmp (panel->model->cmd, PANEL_CMD_DISASMSUMMARY)) {
+			free (tmp);
```

Comment:
i would rather use a bool variable to specify the return and just have the free+return once